### PR TITLE
PIM-5665 : Add Multi Select Attribute Remover

### PR DESCRIPTION
**Pull request by momoss**
| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Y
| Added Behats                      | N
| Changelog updated                 | N
| Review and 2 GTM                  | Y
| Micro Demo to the PO (Story only) | Y
| Migration script                  | N
| Tech Doc                          | N

### CODE REVIEW

File: src/Pim/Component/Catalog/Updater/Remover/MultiSelectAttributeRemover.php [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+        IdentifiableObjectRepositoryInterface $attrOptionRepository,
+        array $supportedTypes
+    ) {
+        parent::__construct($attrValidatorHelper);
+
+        $this->attrOptionRepository = $attrOptionRepository;
+        $this->supportedTypes       = $supportedTypes;
+    }
+
+    public function removeAttributeData(
```
Comment (skeleton) [inline src/Pim/Component/Catalog/Updater/Remover/MultiSelectAttributeRemover.php]: phpdoc

File: src/Pim/Component/Catalog/Updater/Remover/MultiSelectAttributeRemover.php [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+ */
+class MultiSelectAttributeRemover extends AbstractAttributeRemover
+{
+    /** @var IdentifiableObjectRepositoryInterface */
+    protected $attrOptionRepository;
+
+    /**
+     * @param AttributeValidatorHelper              $attrValidatorHelper
+     * @param IdentifiableObjectRepositoryInterface $attrOptionRepository
+     * @param array                                 $supportedTypes
```
Comment (pierallard) [inline src/Pim/Component/Catalog/Updater/Remover/MultiSelectAttributeRemover.php]: string[]

Comment (pierallard) [inline src/Pim/Component/Catalog/Updater/Remover/MultiSelectAttributeRemover.php]: Why $supportedTypes ? I don't see any usage in the class.

File: src/Pim/Component/Catalog/Updater/Remover/MultiSelectAttributeRemover.php [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+
+        $this->removeOptions($product, $attribute, $attributeOptions, $options['locale'], $options['scope']);
+    }
+
+    /**
+     * @param ProductInterface   $product
+     * @param AttributeInterface $attribute
+     * @param array              $attributeOptions
+     * @param string             $locale
+     * @param string             $scope
```
Comment (pierallard) [inline src/Pim/Component/Catalog/Updater/Remover/MultiSelectAttributeRemover.php]: if $locale and $scope can be null, please write string|null

File: src/Pim/Component/Catalog/Updater/Remover/MultiSelectAttributeRemover.php (hunk 2 above)
Comment (momoss) [inline src/Pim/Component/Catalog/Updater/Remover/MultiSelectAttributeRemover.php]: It is used by the AbstractAttributeRemover.

File: src/Pim/Component/Catalog/spec/Updater/Remover/MultiSelectAttributeRemoverSpec.php [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+use Pim\Component\Catalog\Model\AttributeOptionInterface;
+use Pim\Component\Catalog\Exception\InvalidArgumentException;
+
+class MultiSelectAttributeRemoverSpec extends ObjectBehavior
+{
+    function let(
+        AttributeValidatorHelper $attrValidatorHelper,
+        IdentifiableObjectRepositoryInterface $attrOptionRepository
+    )
+    {
```
Comment (pierallard) [inline src/Pim/Component/Catalog/spec/Updater/Remover/MultiSelectAttributeRemoverSpec.php]: merge these lines

File: src/Pim/Component/Catalog/spec/Updater/Remover/MultiSelectAttributeRemoverSpec.php [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+        $attrOptionRepository
+    )
+    {
+        $attribute->getCode()->willReturn('tshirt_style');
+        $attrOptionRepository->findOneByIdentifier('tshirt_style.vneck')->willReturn($attributeOption);
+
+        $product->getValue('tshirt_style', 'fr_FR', 'mobile')->willReturn($productValue);
+
+        $data = ['vneck'];
+        $this->removeAttributeData($product, $attribute, $data, ['locale' => 'fr_FR', 'scope' => 'mobile']);
```
Comment (pierallard) [inline src/Pim/Component/Catalog/spec/Updater/Remover/MultiSelectAttributeRemoverSpec.php]: You have no "should" in this test, what are you checking exactly ?

**Comment by pierallard** [comment]
GTM when comments fixed

File: src/Pim/Component/Catalog/spec/Updater/Remover/MultiSelectAttributeRemoverSpec.php (hunk 5 above)
Comment (momoss) [inline src/Pim/Component/Catalog/spec/Updater/Remover/MultiSelectAttributeRemoverSpec.php]: I forget to add "should"

File: features/enrich/variant-group/display_variant_history.feature [hunk 6]
```diff
@@ -27,13 +27,19 @@ Feature: Display the variant group history
       | 1       | axis     | size              |
 
   Scenario: Successfully edit a variant group attribute and see the change in history
-    And I am on the "caterpillar_boots" variant group page
+    And I am on the variant groups page
```
Comment (phaseinducer) [inline features/enrich/variant-group/display_variant_history.feature]: Given

File: features/enrich/variant-group/display_variant_history.feature [hunk 7]
```diff
@@ -27,13 +27,19 @@ Feature: Display the variant group history
...
+    When I create a new variant group
```
Comment (phaseinducer) [inline features/enrich/variant-group/display_variant_history.feature]: And

File: src/Pim/Component/Catalog/spec/Updater/Remover/MultiSelectAttributeRemoverSpec.php (hunk 4 above)
Comment (phaseinducer) [inline src/Pim/Component/Catalog/spec/Updater/Remover/MultiSelectAttributeRemoverSpec.php]: should be `) {`

File: src/Pim/Component/Catalog/spec/Updater/Remover/MultiSelectAttributeRemoverSpec.php [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+        AttributeOptionInterface $attributeOption,
+        $attrOptionRepository
+    )
+    {
+        $attribute->getCode()->willReturn('tshirt_style');
+        $attrOptionRepository->findOneByIdentifier('tshirt_style.vneck')->willReturn($attributeOption);
+
+        $product->getValue('tshirt_style', 'fr_FR', 'mobile')->willReturn($productValue);
+
+        $data = ['vneck'];
```
Comment (phaseinducer) [inline src/Pim/Component/Catalog/spec/Updater/Remover/MultiSelectAttributeRemoverSpec.php]: useless var

**Comment by phaseinducer** [comment]
GTM after fix

**Pull request merged**